process/utils.go

```diff
@@ -5,10 +5,26 @@ package process
 import (
 	"os"
 	"os/exec"
+	"syscall"
 
+	"github.com/buildkite/agent/logger"
 	"github.com/kr/pty"
 )
 
 func StartPTY(c *exec.Cmd) (*os.File, error) {
 	return pty.Start(c)
 }
+
+func createCommand(name string, arg ...string) *exec.Cmd {
+	return exec.Command(name, arg...)
+}
+
+func terminateProcess(p *os.Process) error {
+	logger.Debug("[Process] Sending signal SIGKILL to PID: %d", p.Pid)
+	return p.Signal(syscall.SIGKILL)
+}
+
+func interruptProcess(p *os.Process) error {
+	logger.Debug("[Process] Sending signal SIGTERM to PID: %d", p.Pid)
+	return p.Signal(syscall.SIGTERM)
```

Comment:
It's weird that this is `SIGTERM` and not `SIGKILL`, but it's how it is currently and I was reluctant to change it here.